stdlib/SuiteSparse/src/spqr.jl

```diff
@@ -342,7 +342,7 @@ end
 _ret_size(F::QRSparse, b::AbstractVector) = (size(F, 2),)
 _ret_size(F::QRSparse, B::AbstractMatrix) = (size(F, 2), size(B, 2))
 
-LinearAlgebra.rank(F::QRSparse) = maximum(F.R.rowval)
+LinearAlgebra.rank(F::QRSparse) = reduce(max, view(F.R.rowval, 1:nnz(F.R)), init = eltype(F.R.rowval)(0))
```

Comment:
Use `SparseArrays.nzvalview` instead?

Reply:
I'm using the `rowval`s here, not the `nzval`s. We could add `rowvalview` but I think we should wait until there are more use cases.